File: 프로그래머스/연습문제/Lv1/소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    vector<bool> primes(n+1, true);
    primes[0] = primes[1] = false;
    
    for(int i=2; i<=n; i++){
        if(!primes[i]) continue;
        for(int j=i*2; j<=n; j+=i)
            primes[j] = false;
    }
    
    int cnt = 0;
    for(int i=2; i<=n; cnt+=primes[i++]);
    
    return cnt;
}

File: 프로그래머스/연습문제/Lv1/소수 찾기/소수 찾기.py
def solution(n):
    primes = [True for i in range(n+1)]
    primes[0] = primes[1] = False

    for i in range(2, n+1):
        if not primes[i]: continue
        for j in range(i*2, n+1, i):
            primes[j] = False

    return primes.count(True)
